Fix mutex ownership

Henrik0x7F opened the pull request:
Fixes #65

Henrik0x7F committed:
Fix mutex ownership

mrexodia commented:
Hm I looked at the code and saw it was originally my code that wasn't working but honestly I cannot say if after this change it's still working because all that multi-threading was very complicated...

Henrik0x7F commented:
Your code worked perfeclty fine. I replaced the execute method with a start, stop and join method. That broke it. The mutex was locked on the calling thread and unlocked on the emulation thread. I changed it so the mutex is unlocked before launching the thread and relocked on the emulation thread. The condition variable blocks the calling thread until the emulation thread has relocked the mutex.
